server.js

```diff
@@ -8,12 +9,63 @@ app.use(cors());
 
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
+const { request, response } = require("express");
 
 app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
+app.get("/bookings", (request, response) => {
+  response.json({ bookings });
+});
+
+app.get("/bookings/:id", (request, response) => {
+  const id = Number(request.params.id);
+  const findId = bookings.find((bookings) => bookings.id === id);
+  if (!findId) {
+    return response
+      .status(404)
+      .json({ message: `booking with the ${id} not found` });
+  }
+  response.status(200).json({ findId })
+});
+
+app.post("/bookings", (request, response) => {
+  if (
+    request.body.firstName === "" ||
+    request.body.email === "" ||
+    request.body.surname == "" ||
+    request.body.title === "" ||
+    request.body.checkInDate === "" ||
+    request.body.checkOutDate === "" ||
+    request.body.roomId === null
+  ) {
+    response.status(400).json({ message: "please fill all the fields" });
+    return;
```

Comment:
this works & 400 makes sense. You could have a think about creating a function to deal with the above (for example the function could take an array of required fields & the request body and return true/false depending on if all required fields are present )